src/lib/pubkey/dilithium/dilithium_common/dilithium.cpp

```diff
@@ -532,7 +532,13 @@ size_t Dilithium_PublicKey::estimated_strength() const {
    return m_public->mode().nist_security_strength();
 }
 
+std::vector<uint8_t> Dilithium_PublicKey::raw_public_key_bits() const {
+   return m_public->raw_pk();
+}
+
 std::vector<uint8_t> Dilithium_PublicKey::public_key_bits() const {
+   // Technically, that's not really correct. The docstring for public_key_bits()
+   // states that it should return a BER-encoding of the public key.
```

Comment:
I think the doc for `public_key_bits` is not really correct/precise. It should return the bytes that, if we stuff it into an X.509 SPKI, result in a valid public key DER encoding. For example for X25519 we just return the 32 byte encoded point with no outer encoding, because that's what RFC 8410 specifies we should do.

Reply:
So the effect of the PQC algorithms returning `raw_public_key_bits` here is that we just stuff the "native" representation of the key as defined by the various reference implementations into the `BIT STRING` of the ASN.1 structure. Which is IMO the correct thing to do, but who knows what will end up getting standardized.

Reply:
I changed this (and the other similar) comments to:

```C++
   // Currently, there isn't a finalized definition of an ASN.1 structure for
   // Dilithium aka ML-DSA public keys. Therefore, we return the raw public key bits.
```

Regarding API stability and "the principle of least surprise": I'm hoping that a finalized implementation of ML-DSA (and the other to-be-standardized algorithms) would return a standardized ASN.1 structure here. While the pre-standard implementations stay with `raw_public_key_bits()` until they are deprecated and eventually phased out. (I'm open for better suggestions.)